### PR TITLE
Add status code support for ASP.NET MVC/WebAPI

### PR DESCRIPTION
**Pull request by kevingosse**

### CODE REVIEW

File: src/Datadog.Trace.ClrProfiler.Managed/Integrations/AspNet/AspNetWebApi2Integration.cs [hunk 1]
```diff
@@ -150,33 +150,53 @@ public static class AspNetWebApi2Integration
...
+            try
             {
-                try
-                {
-                    // call the original method, inspecting (but not catching) any unhandled exceptions
-                    var task = (Task<T>)instrumentedMethod(apiController, controllerContext, cancellationToken);
-                    var responseMessage = await task.ConfigureAwait(false);
+                // call the original method, inspecting (but not catching) any unhandled exceptions
+                var task = (Task<T>)instrumentedMethod(apiController, controllerContext, cancellationToken);
+                var responseMessage = await task;
```
[the remark this answers is not in the document]
Comment (kevingosse) [reply src/Datadog.Trace.ClrProfiler.Managed/Integrations/AspNet/AspNetWebApi2Integration.cs:159]: Yes. We we move out of the synchronization context then we cannot use HttpContext safely

**Comment by kevingosse** [comment]
> LGTM so far. Do we want to add status code to `TracingHttpModule` in this PR or are we doing that separately?

I'm not sure this is needed? This PR works without it.

File: src/Datadog.Trace.ClrProfiler.Managed/Integrations/AspNet/AspNetMvcIntegration.cs [hunk 2]
```diff
@@ -317,16 +317,55 @@ public static Scope CreateScope(object controllerContext)
...
+        private static void SetStatusCode(Scope scope, int statusCode)
+        {
+            try
+            {
+                scope?.Span?.SetTag(Tags.HttpStatusCode, statusCode.ToString());
+            }
+            catch (Exception ex)
             {
-                scope?.Dispose();
+                Log.Error(ex, "Error while setting status code");
```
Comment (lucaspimentel) [inline src/Datadog.Trace.ClrProfiler.Managed/Integrations/AspNet/AspNetMvcIntegration.cs:368]: Here's another one :) 
```suggestion
                Log.Error(ex, "Error while setting span tag with status code");
```

**Comment by lucaspimentel** [comment]
> LGTM so far. Do we want to add status code to `TracingHttpModule` in this PR or are we doing that separately?
> 
> I'm not sure this is needed? This PR works without it.

You're right. I was thinking of Web Forms or other 3rd party web frameworks build on ASP.NET. We still need to add this to `TracingHttpModule` to cover all our web integrations, but that can be another PR.

**Pull request merged**